cmd/nats-blackbox-exporter/main.go

```diff
@@ -0,0 +1,76 @@
+package main
+
+import (
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/nats-io/nats.go"
+	"github.com/snapp-incubator/nats-blackbox-exporter/internal/config"
+	"github.com/snapp-incubator/nats-blackbox-exporter/internal/logger"
+	"go.uber.org/zap"
+)
+
+var publishInterval = 2 * time.Second
+
+func main() {
+	cfg := config.New()
+
+	logger := logger.New(cfg.Logger)
+
+	nc, err := nats.Connect(cfg.NATS.URL)
+	if err != nil {
+		logger.Fatal("nats connection failed", zap.Error(err))
+	}
+
+	logger.Info("nats connection successful",
+		zap.String("connected-addr", nc.ConnectedAddr()),
+		zap.Strings("discovered-servers", nc.DiscoveredServers()))
+
+	nc.SetDisconnectErrHandler(func(_ *nats.Conn, err error) {
+		logger.Fatal("nats disconnected", zap.Error(err))
+	})
+
+	nc.SetReconnectHandler(func(_ *nats.Conn) {
+		logger.Warn("nats reconnected")
+	})
+
+	go subscribe(nc, logger)
+
+	go publish(nc, logger)
+
+	waitForSignal(logger)
+}
+
+func publish(nc *nats.Conn, logger *zap.Logger) {
+	for {
+		err := nc.Publish("subject1", []byte("Hello, NATS!"))
```

Comment:
Use `if err := ...; err != nil` syntax 🙈 . Besides that, NATS has request/response feature ([here](https://pkg.go.dev/github.com/nats-io/nats.go#Conn.Request)) which can be useful here.

Reply:
I use request/response for two-way communication, and I raise an error when no response is received. Should we also consider analyzing one-way (fire-and-forget) communication?

Reply:
Using NATS core, we are testing fire and forget communication. In future, we can test jetstream too. @sinamna do you agree with this?